x/stake/keeper/keeper.go

```diff
@@ -0,0 +1,127 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+// keeper of the stake store
+type Keeper struct {
+	storeKey   sdk.StoreKey
+	cdc        *wire.Codec
+	coinKeeper bank.Keeper
+
+	// codespace
+	codespace sdk.CodespaceType
+}
+
+func NewKeeper(cdc *wire.Codec, key sdk.StoreKey, ck bank.Keeper, codespace sdk.CodespaceType) Keeper {
+	keeper := Keeper{
+		storeKey:   key,
+		cdc:        cdc,
+		coinKeeper: ck,
+		codespace:  codespace,
+	}
+	return keeper
+}
+
+//_________________________________________________________________________
+
+// return the codespace
+func (k Keeper) Codespace() sdk.CodespaceType {
+	return k.codespace
+}
+
+// return the coinkeeper
+func (k Keeper) CoinKeeper() bank.Keeper {
```

Comment:
Do we need to expose this? Means that anyone passed `stake.Keeper` can arbitrarily modify account balances.

Reply:
Yeah I'll modify the structure of the handler to remove this